src/broadcast/metadatabroadcast.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <QObject>
+#include <QLinkedList>
+#include <list>
+
+#include "broadcast/scrobblingservice.h"
+#include "track/track.h"
+#include "track/trackid.h"
+#include "track/trackplaytimers.h"
+
+class MetadataBroadcasterInterface : public QObject {
+    Q_OBJECT
+  public slots:
+    virtual void slotNowListening(TrackPointer pTrack) = 0;
+    virtual void slotAttemptScrobble(TrackPointer pTrack) = 0;
+    virtual void slotAllTracksPaused() = 0;
+  public:
+    virtual ~MetadataBroadcasterInterface() = default;
+    virtual MetadataBroadcasterInterface& 
+        addNewScrobblingService(const ScrobblingServicePtr &newService) = 0;
+    virtual void newTrackLoaded(TrackPointer pTrack) = 0;
+    virtual void trackUnloaded(TrackPointer pTrack) = 0;
+};
+
+class MetadataBroadcaster : public MetadataBroadcasterInterface {
+    Q_OBJECT    
+  private:
+    struct GracePeriod {
+        double m_msElapsed;
+        unsigned int m_numberOfScrobbles = 0;
+        TrackId m_trackId;
+        bool firstTimeLoaded = true;
+        bool hasBeenEjected = false;
+        GracePeriod(double msElapsed,TrackPointer pTrack) :
+        m_msElapsed(msElapsed),m_trackId(pTrack->getId()) {}
+        bool operator==(const GracePeriod &other) const {
```

Comment:
When is this used? It is not a fully compare of al elements so it requires some comments.

Reply:
I think it is better to use a QHash with a m_trackId as key. That makes the code more explicit.